sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/ClientRetryPolicy.java

```diff
@@ -152,6 +160,33 @@ public Mono<ShouldRetryResult> shouldRetry(Exception e) {
         return this.throttlingRetry.shouldRetry(e);
     }
 
+      private boolean gatewayRequestCanFailoverOnTimeout(RxDocumentServiceRequest request, CosmosException clientException) {
+        //Query Plan requests
+        if(request.getResourceType() == ResourceType.Document
```

Comment:
I wonder if we really need to check for ResourceType here? Because we have always relied on the fact that QueryPlan calls are only for documents, so at other places we only check for operationType == QueryPlan

Reply:
Or even if QueryPlan calls are not for documents only, would we not want to retry on other region?